__tests__/reducers/windows.spec.ts

```diff
@@ -44,16 +45,74 @@ describe( 'windows reducer', () =>
                         closedtabs : [],
                         lastActiveTabs : []
                     }
-                  }
+                }
+            });
+        });
+        it('should handle adding another window',()=>{
+            const state =
+            {
+                openWindows: 
+                {
+                    '1' : basicwindow
+                },
+                closedWindows : 
+                {
+                    '1': 
+                    {
+                        closedtabs : [],
+                        lastActiveTabs : []
+                    }
+                }
+            }
+            const addWindow = windows(JSON.parse(JSON.stringify(state)),{
+                type: TYPES.ADD_WINDOW,
+                payload : { windowId: 2 }
+            });
+            expect(
+                addWindow
+            ).toEqual({
+                openWindows: 
+                {
+                    '1' : basicwindow,
+                    '2' : basicwindow
+                },
+                closedWindows : 
+                {
+                    '1': 
+                    {
+                        closedtabs : [],
+                        lastActiveTabs : []
+                    },
+                    '2': 
+                    {
+                        closedtabs : [],
+                        lastActiveTabs : []
+                    }
+                }
             });
         });
     });
     describe('ADD_TAB_NEXT',()=>{
-        const addTabNext = windows(JSON.parse(JSON.stringify(addWindow)),{
-            type: TYPES.ADD_TAB_NEXT,
-            payload : { windowId: 1, tabId }
-        });
         it('should add handle add tab without passing index',()=>{
+            const state = 
+            {
+                openWindows: 
+                {
+                    '1' : basicwindow
+                },
+                closedWindows : 
+                {
+                    '1': 
+                    {
+                        closedtabs : [],
+                        lastActiveTabs : []
+                    }
+                }
+            }
+            const addTabNext = windows(JSON.parse(JSON.stringify(state)),{
```

Comment:
You're creating a new state just above. I don't think the parse/stringify combo is needed here...?